src/utilities.py

```diff
@@ -210,9 +211,12 @@ def set_attributes(new_object, template):
     for attribute in template.__dict__:
         if not hasattr(new_object, attribute):
             setattr(new_object, attribute, getattr(template, attribute))
+    delete_attributes = []
     for attribute in new_object.__dict__:
         if not hasattr(template, attribute):
-            delattr(new_object, attribute)
+            delete_attributes.append(attribute)
```

Comment:
Instead of using
```python
 delete_attributes = []
    for attribute in new_object.__dict__:
        if not hasattr(template, attribute):
            delete_attributes.append(attribute)
```
I'd use
```python
delete_attributes = [attribute for attribute in new_object.__dict__ if not hasattr(template, attribute)]
```